processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/vector/VectorGroupByEngine.java

```diff
@@ -138,12 +138,22 @@ public void close()
             try {
               final VectorColumnSelectorFactory columnSelectorFactory = cursor.getColumnSelectorFactory();
               final List<GroupByVectorColumnSelector> dimensions = query.getDimensions().stream().map(
-                  dimensionSpec ->
-                      ColumnProcessors.makeVectorProcessor(
+                  dimensionSpec -> {
+                    if (dimensionSpec instanceof DefaultDimensionSpec) {
```

Comment:
It's a defensive move in case any `DimensionSpec` other than `DefaultDimensionSpec` are vectorizable in the future. I don't expect this to happen either, but it didn't sit right to write code that would be incorrect for other `DimensionSpec` types.